mirgecom/profiling.py

```diff
@@ -168,18 +149,57 @@ def _finish_profile_events(self) -> None:
 
             self.profile_results.setdefault(program, []).append(new)
 
-        for t in nonloopy_profile_events:
-            program = t.knl
-            time = t.cl_event.profile.end - t.cl_event.profile.start
-            new = ProfileResult(time, t.nops, t.nbytes, t.nbytes)
-            self.profile_results.setdefault(program, []).append(new)
-
         self.profile_events = []
-        nonloopy_profile_events = []
 
-    def tabulate_profiling_data(self) -> pytools.Table:
+    def get_profiling_data_for_kernel(self, kernel_name: str,
+                                 wait_for_events=True) -> ProfileResultsForKernel:
+        """Return value of profiling result for kernel `kernel_name`."""
+        if wait_for_events:
+            self._finish_profile_events()
+
+        def _gather_data(results_list: list):
```

Comment:
That type annotation seems wrong.
```
del results_list[key]
```
wouldn't work on a list.

(And if it's not a list, it shouldn't be called "list" either.)

Reply:
Fixed.